clang/lib/Driver/ToolChains/AVR.cpp

```diff
@@ -400,6 +400,14 @@ void AVRToolChain::AddClangSystemIncludeArgs(const ArgList &DriverArgs,
 void AVRToolChain::addClangTargetOptions(
     const llvm::opt::ArgList &DriverArgs, llvm::opt::ArgStringList &CC1Args,
     Action::OffloadKind DeviceOffloadKind) const {
+  // Reject C/C++ compilation for avr1 devices.
+  const Driver &D = getDriver();
+  std::string CPU = getCPUName(D, DriverArgs, getTriple());
+  std::optional<StringRef> FamilyName = GetMCUFamilyName(CPU);
+  if (CPU == "avr1" || (FamilyName && FamilyName->compare("avr1") == 0))
+    D.Diag(diag::err_drv_opt_unsupported_input_type) << "avr1"
+                                                     << "c/c++";
```

Comment:
The error is certainly the right category, but it's a shame you can't have it say "device family 'avr1'" (replace "device family" with whatever the correct term is).

I looked around and I don't see a way to use an existing error but prepend to the text. Only other way is to make a whole new error type which seems a shame when this is so close. Not that big of a deal, but it would help new users I think.

@Sirraide you helped me on diagnostics before, is it plausible to do that or is a new error type the only way?

Reply:
You could instead put `-mmcu=attiny11` as the value, then users know what part of the command triggered the issue, though they may not be aware that this whole category is the same. In practice I wonder if that would be more useful for most people.

Also what does gcc say in this situation?

Reply:
> You could instead put -mmcu=attiny11 as the value

Assuming that you know that `-mmcu` is the only way to hit this code path.

Reply:
Since `-mmcu` is the only way to trigger this issue, I have updated my error message to `'-mmcu=attiny11' invalid for input of type c/c++`